core/src/main/java/org/mskcc/cbio/portal/servlet/PathologyReportView.java

```diff
@@ -0,0 +1,224 @@
+
+    // class which process access control to cancer studies
+    private AccessControl accessControl;
+
+    // root directory of internal pathology reports or null if undefined
+    private static final String DATA_DIRECTORY = GlobalProperties.getInternalPathReportRoot();
+    
+    /**
+     * Initializes the servlet.
+     *
+     * @throws ServletException Serlvet Init Error.
+     */
+    @Override
+    public void init() throws ServletException {
+        super.init();
+		accessControl = SpringUtil.getAccessControl();
+    }
+    
+    /** 
+     * Processes requests for both HTTP <code>GET</code> and <code>POST</code> methods.
+     * @param request servlet request
+     * @param response servlet response
+     * @throws ServletException if a servlet-specific error occurs
+     * @throws IOException if an I/O error occurs
+     */
+    protected void processRequest(HttpServletRequest request, HttpServletResponse response)
+        throws ServletException, IOException {        
+        try {            
+            validateRequest(request);
+            if (request.getAttribute(ERROR_CODE)!=null) {
+                response.sendError(Integer.parseInt((String) request.getAttribute(ERROR_CODE)), 
+                    (String) request.getAttribute(ERROR_MSG));
+            } else {
+                File pathologyReport = getRequestedFile(request);
+                serveFile(response, pathologyReport);
+            }
+        } catch (DaoException e) {
+            logger.error("Got Database Exception while processing request for pathology report.", e);
+            response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR, 
+                "An error occurred while trying to connect to the database.");
+        } catch (NoSuchFileException e) {
+            logger.error("Got No Such File Exception while processing request for pathology report.", e);
+            response.sendError(HttpServletResponse.SC_NOT_FOUND, 
+                "The requested pathology report could not be located.");
+        }
+    }
+    
+    private String getRequestedPath(HttpServletRequest request) throws IOException{
+        return URLDecoder.decode(request.getPathInfo().substring(1), "UTF-8");
+    }
+    
+    private File getRequestedFile(HttpServletRequest request) throws IOException {
+        // Convert from the expected url path 'studyId/patientId/sampleId.pdf' to the expected filesystem path 'studyId.patientId.sampleId.pdf'
+        final String convertedPath = getRequestedPath(request).replace("/", ".");
+        return new File(DATA_DIRECTORY, convertedPath);
+    }
+    
+    private void setError(HttpServletRequest request, int httpStatusCode, String errorMessage) {
+        request.setAttribute(ERROR_CODE, Integer.toString(httpStatusCode));
+        request.setAttribute(ERROR_MSG, errorMessage);
+    }
+    
+    private boolean validateRequest(HttpServletRequest request) throws IOException, DaoException {
```

Comment:
It doesn't need to since there is a check on line 97 to see if the validation raised any errors. It could just be changed to a void method with early returns (rather than return true/false), or alternatively line 97 could be modified to do something with the boolean return value (such as if validate == true, then serve expected file, otherwise check for errors).
What do you suggest?